src/ameba/rule/lint/unused_expression.cr

```diff
@@ -0,0 +1,157 @@
+module Ameba::Rule::Lint
+  # A rule that disallows unused expressions.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # a = obj.method do |x|
+  #   x == 1 # => Comparison operation has no effect
+  #   puts x
+  # end
+  #
+  # Float64 | StaticArray(Float64, 10)
+  #
+  # pointerof(foo)
+  # ```
+  #
+  # And these are considered valid:
+  #
+  # ```
+  # a = obj.method do |x|
+  #   x == 1
+  # end
+  #
+  # foo : Float64 | StaticArray(Float64, 10) = 0.1
+  #
+  # bar = pointerof(foo)
+  # ```
+  #
+  # This rule currently supports checking for unused:
+  # - comparison operators: `<`, `>=`, etc.
+  # - generics and unions: `String?`, `Int32 | Float64`, etc.
+  # - literals: strings, bools, chars, hashes, arrays, range, etc.
+  # - pseudo-method calls: `sizeof`, `is_a?` etc.
+  # - variable access: `local`, `@ivar`, `@@cvar`, `self`
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/UnusedExpression:
+  #   Enabled: true
+  # ```
+  class UnusedExpression < Base
+    properties do
+      since_version "1.7.0"
+      description "Disallows unused expressions"
+    end
+
+    def test(source : Source)
+      AST::ImplicitReturnVisitor.new(self, source)
+    end
+
+    CLASS_VAR_MSG = "Value from class variable access is unused"
+
+    def test(source, node : Crystal::ClassVar, in_macro : Bool)
+      # Class variables aren't supported in macros
+      return if in_macro
+
+      issue_for node, CLASS_VAR_MSG
+    end
+
+    COMPARISON_MSG       = "Comparison operation is unused"
```

Comment:
Please group the constants together, splitting them to ones containing issue messages and others.
Also, please stick to `MSG_XXX` format.